### DISCUSSION TITLE
Request Attributes and Containers

### DISCUSSION THREAD

**Discussion by TroyDavidson** [brefphp/bref#1237] [General]
I have to use an AWS Lambda container for my code.  The problem I'm having is getting my url attributes parsed out and into my Handler.

I have my url defined as this in serverless.yml:
`http: "GET /client/{cid}/payment/{id}"`

In my code that calls the Handler, I'm building a ServerRequest object and then pass it to my Handler:
```
$request = new ServerRequest($_SERVER['REQUEST_METHOD'], $_SERVER['REQUEST_URI'], getallheaders(), file_get_contents('php://input'));
$response = $queueDispatcher->handle($request);
```

But, when I parse out the attributes, I don't see **cid** nor **id** in the array:
`$this->logger->info("attributes:" . print_r($request->getAttributes(), true));
`

Am I doing something wrong, or should I be doing this another way?

**Comment by dhrrgn** [comment]
How you get the path parameters depends on What type of function you're using. I assume you are using the FPM runtime, but are trying to get the parameters like you do with the Event-Based runtime. Here's more details.

**FPM Runtime (`php-XX-fpm` layers)** 

Bref, sadly, does not pass along the path parameters from the API Gateway event. You can see how it [converts the event to a FastCGI request here](https://github.com/brefphp/bref/blob/master/src/Event/Http/FpmHandler.php#L217-L244).

You must using a router of some sort, or otherwise parse the URL yourself, just like if it was a standard web request.

**Event-Based Runtime (`php-XX` layers)**

Using a [PSR-15 request handler](https://bref.sh/docs/function/handlers.html#api-gateway-http-events) you can access the parameters through the `ServerRequestInterface` object passed to the handler. 

Hope that helps!
